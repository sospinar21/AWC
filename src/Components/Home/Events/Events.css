.main-banner {
  background-repeat: no-repeat;
  background-size: cover;
  margin: 5vh auto 5vh;
  width: 100%;
  background-position-y: -10vh;
  height: 45vh;
  background-image: url("https://2.bp.blogspot.com/-yE3M4nNXZ58/WMbInF32VBI/AAAAAAAACeA/uiXXIXrIvbYPD4CaaGC6tGDR7eWPtdTlgCLcB/s1600/Dido%2Bimage%2Bfor%2BFB.jpg"); }

.box2 {
  border-radius: 3px;
  border-style: solid;
  border-color: rgba(0, 0, 0, 0.05);
  border-image: initial; }

.suggEvent-cards {
  width: 20%;
  margin: 2.3vh;
  height: 15vh;
  overflow-y: hidden;
  height: 35vh;
  border-radius: 5%; }
  .suggEvent-cards img {
    width: 100%;
    margin-bottom: 1vh; }
  .suggEvent-cards h4 {
    font-size: 2.3vh;
    margin: 1.5vh auto;
    color: purple;
    letter-spacing: 0.5px; }
  .suggEvent-cards p {
    letter-spacing: .3px;
    font-family: 'Lato', sans-serif;
    font-weight: inherit;
    color: dimgrey;
    ine-height: 1.5; }

.cards-container {
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
  margin-top: 2vh; }

.description {
  margin-top: 1vh;
  overflow-y: hidden; }

.black-bgn {
  width: 80%;
  height: inherit;
  background-color: #00000045;
  position: absolute;
  z-index: 2; }

.main-banner-text {
  z-index: 4;
  margin: 9vh;
  width: 40%;
  padding: 5vh;
  color: white !important;
  background-color: #0000008a; }
  .main-banner-text h1 {
    font-size: 5vh;
    margin: 1vh; }
  .main-banner-text h2 {
    font-size: 4vh; }

.events-small-box {
  height: 12vh;
  width: 100%;
  display: -ms-flexbox;
  display: flex;
  text-align: left;
  border-bottom: 1px solid #ddd;
  overflow: hidden;
  border-radius: 5px;
  padding-bottom: 20px;
  margin-bottom: 20px; }
  .events-small-box img {
    height: 100%;
    margin-right: 1.7vh;
    border-radius: 5px;
    -webkit-box-shadow: 1px 2px 3px #ccc;
    box-shadow: 1px 2px 3px #ccc; }
  .events-small-box h4 {
    font-size: 1rem;
    margin-left: 12px;
    margin: 0; }
