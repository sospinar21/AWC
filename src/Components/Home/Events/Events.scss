.main-banner {
  background-repeat: no-repeat;
  background-size: cover;
  margin: 50px auto 50px;
  width: 100%;
  background-position-y: -100px;
  height: 450px;
  background-image: url('https://2.bp.blogspot.com/-yE3M4nNXZ58/WMbInF32VBI/AAAAAAAACeA/uiXXIXrIvbYPD4CaaGC6tGDR7eWPtdTlgCLcB/s1600/Dido%2Bimage%2Bfor%2BFB.jpg')
}

.box2 {
  border-radius: 3px;
  border-style: solid;
  border-color: rgba(0, 0, 0, 0.05);
  border-image: initial;
}

  .content-e{
    text-align: right;
  }

.event-b{
  font-size: 17px;
}

.suggEvent-cards {
  width: 20%;
  margin: 20.3px;
  height: 150px;
  overflow-y: hidden;
  height: 350px;
  border-radius: 5%;

  img {
    width: 100%;
    margin-bottom: 10px;
  }

  h4 {
    font-size: 20.3px;
    margin: 10.5px auto;
    color: purple;
    letter-spacing: 0.5px;
  }

  p {
    letter-spacing: .3px;
    font-family: 'Lato', sans-serif;
    font-weight: inherit;
    color: dimgrey;
    
  }
}

.cards-container {
  display:flex;
  justify-content: space-around;
  flex-wrap: wrap;
  margin-top: 20px;
}

.description {
  margin-top: 10px;
  overflow-y: hidden;
}

.black-bgn {
  width: 80%;
  height: inherit;
  background-color: #00000045;
  position: absolute;
  z-index: 2;
}

.main-banner-text{
  z-index: 4;
  margin: 90px;
  width: 40%;
  padding: 50px;
  color: white !important;
  background-color: #0000008a;

  h1 {
    font-size: 50px;
    margin: 10px
  }

  h2 {
    font-size: 40px    
  }
}

.content-e img{
  max-height: 100px;
}

.events-small-box {
  height: 95px;
  width: 100%;
  display: -ms-flexbox;
  display: flex;
  text-align: left;
  border-bottom: 1px solid #ddd;
  overflow: hidden;
  border-radius: 5px;
  padding-bottom: 20px;
  margin-bottom: 20px;
  cursor: pointer;

  img {
    height: 100%;
    margin-right: 10.7px;
    border-radius: 5px;
    -webkit-box-shadow: 1px 2px 3px #ccc;
    box-shadow: 1px 2px 3px #ccc;
  }

  h4 {
    font-size: 14px;
    margin-left: 12px;
    margin: 0;
  }
}

.events-small-box:hover{
  background-color: #d3d3d33b  
}

.main-events {
  height: 300px;
  overflow-y: scroll;
}
