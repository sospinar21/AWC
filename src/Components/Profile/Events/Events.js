import React, { Component } from 'react'

export class Events extends Component {
  



  render () {
    return()
  }
}

export mapStateToProps = {}

export mapDispatchToProps = () => ({})

export connect(mapStateToProps, mapDispatchToProps)(Events)