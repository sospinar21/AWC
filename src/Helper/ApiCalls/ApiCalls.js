import React, {Component} from 'react';

class ApiCalls extends Component {
  constructor(){
    super()
  }

  fetchEvents = async () => {
    const url = `https://api.awc.dance/events?city=vermont`;
    const response = await fetch(url)
    const data= await response.json();
    return data;
  } 

  fetchStudios = () => {
    const url = `https://api.awc.dance/?city=80202`;
    return fetch(url, {
      method: 'GET',
      headers: {
        'Content-type': 'application/json',
      }
    }).then( response => response.json() )
      .then( data => data )   
  }

  fetchVideos= async () => {
    var url = 'https://api.awc.dance/youtubevideos'
    const response = await fetch(url)
    const data = await response.json()
    return data
  }
}

export default ApiCalls;