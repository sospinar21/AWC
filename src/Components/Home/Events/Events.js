import React, { Component } from 'react'
import  ApiCalls  from '../../../Helper/ApiCalls/ApiCalls'
import { connect } from 'react-redux';
import { addEvents } from '../../../Actions/actions'
import './Events.css'

export class Events extends Component {
  constructor () {
    super ()
    
  }

  componentDidMount() {
    this.fetchEvents()
  }
  
  fetchEvents = async () => {
    let api = new ApiCalls()
    const suggestedEvents = await api.fetchEvents()
    await this.props.addEvents(suggestedEvents)
  }

  displayEvents = () => {
    const events = this.props.suggestedEvents.map(suggEvent => {
      if (suggEvent.logo){
        var img = suggEvent.logo.url
      } else {
        var img = 'https://www.kent.edu/sites/default/files/styles/teaser_image/public/page/B0B_4055crop.JPG?itok=4ie7uvK-'
      }
      return (
        <div key={suggEvent.id} className='suggEvent-cards'>
          <img src={img} />
          <div className='text-container'>
            <h4>{suggEvent.name}</h4>
            <b>{suggEvent.date.utc}</b>
            <div className='description'>
              <p>{suggEvent.description}</p>
            </div>
          </div>   
        </div>
      )
    })
    return events;
  }

  eventsInMain = () => {
    const events = this.props.suggestedEvents.map(suggEvent => {
      if (suggEvent.logo){
        var img = suggEvent.logo.url
      } else {
        var img = 'https://www.kent.edu/sites/default/files/styles/teaser_image/public/page/B0B_4055crop.JPG?itok=4ie7uvK-'
      } 
      return (
        <div key={suggEvent.id} className='events-small-box'>
          <img src={img} />
          <h4>{suggEvent.name}</h4>
        </div>
      )
    })
    return events;
  }

  render () {
    if (this.props.suggestedEvents.length) {
    //   return (
    //     <div>
    //       <h1>Popular Events</h1>          
    //       <div className='main-banner box2'> 
    //         <div className='black-bng'/>
    //         <div className = 'main-banner-text'>
    //           <h1>10th Anniversary</h1>
    //           <h2>Learn More</h2>
    //         </div>
    //       </div>
    //       <h1>Local Events</h1>
    //       <div className='cards-container'>
    //         {this.displayEvents()}
    //       </div>
    //     </div>
    //   )
    // }
    // else {
      return (
        <div> 
          <div className='cards-container-small'>
            {this.eventsInMain()}
          </div>
        </div>
      )
    }
    else {
      return (
        <h1>Loading </h1>
      )
    }
  }
}

export const mapStateToProps = (state) => {
  return ({
    suggestedEvents: state.suggestedEvents
  })
}

export const mapDispatchToProps = dispatch => ({
  addEvents: (eventsData) => dispatch(addEvents(eventsData)) 
  
})

export default connect(mapStateToProps, mapDispatchToProps)(Events)