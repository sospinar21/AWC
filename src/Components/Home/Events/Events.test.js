import { Events, mapStateToProps, mapDispatchToProps } from '../Events/Events'
import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from 'enzyme';
import  ApiCalls  from '../../../Helper/ApiCalls/ApiCalls';


describe('events', () => {
  let events;
  let mockProps;
  let api = new ApiCalls();

  beforeEach(() => {

    mockProps = {
      suggestedEvents: [{name: 'party'}],
      addEvents: jest.fn(),
      selectedEvent: {name:'workshop'},
      addSelectedEvent: jest.fn(),
      selectedLocation: 'Denver'
    }

    api.fetchEvents = jest.fn()
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      status: 200,
      json: () => Promise.resolve({data: {}})
    }));
    events = shallow(<Events {...mockProps} /> )

  })


  it('matches snapshot', () => {
    expect(events).toMatchSnapshot();
  })

  describe('componentDidMount', () => {
    it('calls fetchEvents', () => {
      events = mount(<Events {...mockProps} /> )

      events.instance().componentDidMount()
      events.instance().fetchEvents = jest.fn()
      expect(events.instance().fetchEvents).toHaveBeenCalled()
    })
  })

  describe('fetchEvents', () => {

    it('calls window.fetch with the correct params', async () => {
      const expected = ["https://api.awc.dance/events?city=Denver"]

      await events.instance().fetchEvents(mockProps.selectedLocation)
      expect(window.fetch).toHaveBeenCalledWith(expected)
    })

    it('calls props.addEvents', () => {

      events.instance().fetchEvents()
      expect(mockProps.addEvents).toHaveBeenCalledWith({"data": {}})
    })
    
  })

  describe('eventsInMain', () => {

    it('shows a default image if the event doesnt have one', () => {

      events.instance().eventsInMain()
      
      const logo = events.find('img')

      expect(logo.props('src').value).toEqual(undefined)
    })
    
  })

  describe('eventsinMain', () => {
    
  })

  describe('mapStateToProps', () => {
    
    let mockDispatch;
    let suggestedEvents;
    let mappedProps

    it('returns an object with the suggestedEvents', () => {

      const mockState = {
        suggestedEvents: {name: 'Workshop'}
      }

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(mockState);
      
    })
  })

  describe('mapDispatchToProps', () => {

    it('should call dispatch with the correct params', () => {
    
      let mockDispatch = jest.fn();
      let mappedProps = mapDispatchToProps(mockDispatch);
      
      const eventsData = {id: 1, title:'workshop'}      
      const mockAction = {
        type: 'ADD_EVENTS',
        eventsData
      };
  
      mappedProps.addEvents(eventsData);
      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    })

  })
  
})
