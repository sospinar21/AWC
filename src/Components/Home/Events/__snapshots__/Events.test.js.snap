// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`events matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Events
    addEvents={[Function]}
    addSelectedEvent={[Function]}
    selectedEvent={
        Object {
            "name": "workshop",
          }
    }
    selectedLocation="Denver"
    suggestedEvents={
        Array [
            Object {
              "name": "party",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="cards-container-small"
>
        <div
                className="events-small-box"
                onClick={[Function]}
        >
                <img
                        src="https://www.kent.edu/sites/default/files/styles/teaser_image/public/page/B0B_4055crop.JPG?itok=4ie7uvK-"
                />
                <h4>
                        party
                </h4>
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="events-small-box"
            onClick={[Function]}
>
            <img
                        src="https://www.kent.edu/sites/default/files/styles/teaser_image/public/page/B0B_4055crop.JPG?itok=4ie7uvK-"
            />
            <h4>
                        party
            </h4>
</div>,
        ],
        "className": "cards-container-small",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "[object Object]0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                src="https://www.kent.edu/sites/default/files/styles/teaser_image/public/page/B0B_4055crop.JPG?itok=4ie7uvK-"
/>,
              <h4>
                party
</h4>,
            ],
            "className": "events-small-box",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "src": "https://www.kent.edu/sites/default/files/styles/teaser_image/public/page/B0B_4055crop.JPG?itok=4ie7uvK-",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "party",
              },
              "ref": null,
              "rendered": "party",
              "type": "h4",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="cards-container-small"
>
          <div
                    className="events-small-box"
                    onClick={[Function]}
          >
                    <img
                              src="https://www.kent.edu/sites/default/files/styles/teaser_image/public/page/B0B_4055crop.JPG?itok=4ie7uvK-"
                    />
                    <h4>
                              party
                    </h4>
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="events-small-box"
              onClick={[Function]}
>
              <img
                            src="https://www.kent.edu/sites/default/files/styles/teaser_image/public/page/B0B_4055crop.JPG?itok=4ie7uvK-"
              />
              <h4>
                            party
              </h4>
</div>,
          ],
          "className": "cards-container-small",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "[object Object]0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  src="https://www.kent.edu/sites/default/files/styles/teaser_image/public/page/B0B_4055crop.JPG?itok=4ie7uvK-"
/>,
                <h4>
                  party
</h4>,
              ],
              "className": "events-small-box",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "src": "https://www.kent.edu/sites/default/files/styles/teaser_image/public/page/B0B_4055crop.JPG?itok=4ie7uvK-",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "party",
                },
                "ref": null,
                "rendered": "party",
                "type": "h4",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
